magicblock-ledger/src/store/api.rs

```diff
@@ -1005,21 +1003,21 @@ impl Ledger {
         )>,
     > + '_ {
         let (_lock, _) = self.ensure_lowest_cleanup_slot();
-        let iterator_mode = iterator_mode.unwrap_or(IteratorMode::Start);
+        let from = from.unwrap_or((Signature::default(), 0));
         self.transaction_status_cf
-            .iter_protobuf(iterator_mode)
+            .iter_protobuf(IteratorMode::From(from, IteratorDirection::Forward))
```

Comment:
Always starting at slot 0 makes this a lot slower.
The previous implementation would start at a specific slot.

Once the ledger is very large this would take a very long time. I'm not sure this should be needed.